### PR TITLE
Implement boards update / delete / create

### LINKED ISSUE #2: Boards Endpoints
- [x] Create a Board
- [x] Delete a Board
- [x] Edit a Board
- [x] Retrieve information about a Board

### PR DESCRIPTION
**Pull request by BrandonRomano**
Fixes #2

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -157,3 +158,216 @@ func (suite *ClientTestSuite) TestUnauthorizedBoardFetch() {
...
+		assert.Equal(suite.T(), http.StatusUnauthorized, pinterestError.StatusCode)
+	} else {
+		// Make this error out, should always be a PinterestError
+		assert.Equal(suite.T(), true, false)
+	}
+}
+
+// TestForbiddenBoardUpdate tests that you may not update boards
+// that do not belong to you.
+func (suite *ClientTestSuite) TestForbiddenBoardUpdate() {
```
Comment (BrandonRomano) [inline client_test.go:234]: Should rename this method to UnauthorizedBoardUpdate... As actually a 401 is returned

File: client_test.go [hunk 2]
```diff
@@ -157,3 +158,216 @@ func (suite *ClientTestSuite) TestUnauthorizedBoardFetch() {
...
+		assert.Equal(suite.T(), http.StatusNotFound, pinterestError.StatusCode)
+	} else {
+		// Make this error out, should always be a PinterestError
+		assert.Equal(suite.T(), true, false)
+	}
+}
+
+// TestNotFoundBoardDelete tests that an error is appropriately thrown
+// when trying to delete a board that does not belong to the user
+func (suite *ClientTestSuite) TestForbiddenBoardDelete() {
```
Comment (BrandonRomano) [inline client_test.go:320]: again not forbidden here...

**Comment by BrandonRomano** [comment]
Pinterest is actually totally incorrectly throwing around the 401s, and I picked the right naming for the `Forbidden` cases... I'm going to keep the naming the same, as I will run into collisions with my actual real 401 checks!

File: controllers/boards_controller.go [hunk 3]
```diff
@@ -19,13 +19,84 @@ func NewBoardsController(wc *wrecker.Wrecker) *BoardsController {
...
 }
 
-// Fetch loads a board from the board's username / boardName.
-func (bc *BoardsController) Fetch(username string, boardName string) (*models.Board, error) {
+// Fetch loads a board from the board_spec (username/board-slug)
+func (bc *BoardsController) Fetch(boardSpec string) (*models.Board, error) {
+	// Make request
 	response := new(models.Response)
 	response.Data = new(models.Board)
+	resp, err := bc.wreckerClient.Get("/boards/"+boardSpec).
```
Comment (hhsnopek) [inline controllers/boards_controller.go:27]: should have a space between operator

File: controllers/boards_controller.go [hunk 4]
```diff
@@ -45,3 +116,29 @@ func (bc *BoardsController) Fetch(username string, boardName string) (*models.Bo
 	// OK
 	return response.Data.(*models.Board), nil
 }
+
+// Delete deletes an existing board
+func (bc *BoardsController) Delete(boardSpec string) error {
+	// Make request
+	response := new(models.Response)
+	response.Data = ""
+	resp, err := bc.wreckerClient.Delete("/boards/" + boardSpec + "/").
```
Comment (hhsnopek) [inline controllers/boards_controller.go:125]: reference to the operator with space (concerning previous comment)

**Pull request merged**